packages/golden-layout/src/container/ItemContainer.ts

```diff
@@ -146,14 +151,32 @@ export default class ItemContainer<
    * Closes the container if it is closable. Can be called by
    * both the component within at as well as the contentItem containing
    * it. Emits a close event before the container itself is closed.
+   * @param options Options to pass into the beforeClose handler
    */
-  close() {
-    if (this._config.isClosable) {
+  close(options?: CloseOptions) {
```

Comment:
This method could be rewritten to be clearer; most notably, we don't need to do anything if `isClosable` is false (no need to check beforeCloseHandler). Could write something like

```
if (!this._config.isClosable) return;

if (this.beforeCloseHandler?.(options) ?? true) {
  this.emit('close');
  this.parent.close();
}
```